payloads/v9/channel.ts

```diff
@@ -0,0 +1,1047 @@
+ */
+export interface APISticker {
+	/**
+	 * ID of the sticker
+	 */
+	id: Snowflake;
+	/**
+	 * ID of the pack the sticker is from
+	 */
+	pack_id: Snowflake;
+	/**
+	 * Name of the sticker
+	 */
+	name: string;
+	/**
+	 * Description of the sticker
+	 */
+	description: string;
+	/**
+	 * A comma-separated list of tags for the sticker
+	 */
+	tags?: string;
+	/**
+	 * Sticker asset hash
+	 */
+	asset: string;
+	/**
+	 * Type of sticker format
+	 *
+	 * See https://discord.com/developers/docs/resources/channel#message-object-message-sticker-format-types
+	 */
+	format_type: StickerFormatType;
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#message-object-message-sticker-format-types
+ */
+export const enum StickerFormatType {
+	PNG = 1,
+	APNG,
+	LOTTIE,
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#followed-channel-object
+ */
+export interface APIFollowedChannel {
+	/**
+	 * Source channel id
+	 */
+	channel_id: Snowflake;
+	/**
+	 * Created target webhook id
+	 */
+	webhook_id: Snowflake;
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#reaction-object-reaction-structure
+ */
+export interface APIReaction {
+	/**
+	 * Times this emoji has been used to react
+	 */
+	count: number;
+	/**
+	 * Whether the current user reacted using this emoji
+	 */
+	me: boolean;
+	/**
+	 * Emoji information
+	 *
+	 * See https://discord.com/developers/docs/resources/emoji#emoji-object
+	 */
+	emoji: APIPartialEmoji;
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#overwrite-object-overwrite-structure
+ */
+export interface APIOverwrite {
+	/**
+	 * Role or user id
+	 */
+	id: Snowflake;
+	/**
+	 * Either 0 (role) or 1 (member)
+	 *
+	 * {@link OverwriteType}
+	 */
+	type: OverwriteType;
+	/**
+	 * Permission bit set
+	 *
+	 * See https://discord.com/developers/docs/topics/permissions#permissions-bitwise-permission-flags
+	 *
+	 * See https://en.wikipedia.org/wiki/Bit_field
+	 */
+	allow: Permissions;
+	/**
+	 * Permission bit set
+	 *
+	 * See https://discord.com/developers/docs/topics/permissions#permissions-bitwise-permission-flags
+	 *
+	 * See https://en.wikipedia.org/wiki/Bit_field
+	 */
+	deny: Permissions;
+}
+
+export const enum OverwriteType {
+	Role,
+	Member,
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#thread-metadata-object-thread-metadata-structure
+ */
+export interface APIThreadMetadata {
+	/**
+	 * The archived state of the thread
+	 */
+	archived: boolean;
+	/**
+	 * The id of the member that last changed `archived`
+	 */
+	archiver_id?: Snowflake;
+	/**
+	 * The duration to wait before automatically archive the thread
+	 */
+	auto_archive_duration: ThreadAutoArchiveDuration;
+	/**
+	 * An ISO8601 timestamp for when `archived` was last changed
+	 */
+	archive_timestamp: string;
+	/**
+	 * The locked state of the thread
+	 */
+	locked?: boolean;
+}
+
+export type ThreadAutoArchiveDuration = 60 | 1440 | 4320 | 10080;
```

Comment:
let's make this an enum instead, easier for the end users

Reply:
do you have a suggested naming scheme for this?

Reply:
See https://github.com/discordjs/discord-api-types/pull/133#issuecomment-846378899 since GH really did a good job placing it in the right spot

Reply:
Okay, done, the client calls 1 day 24 hours, but enum names can't start with a number, so I did OneDay, do you think TwentyFourHours would be better.

Reply:
> Okay, done, the client calls 1 day 24 hours, but enum names can't start with a number, so I did OneDay, do you think TwentyFourHours would be better.

Nahhhh, OneDay is fine! TwentyFourHours is just a mouthful